js/views/modals/purchase/Purchase.js

```diff
@@ -252,10 +252,18 @@ export default class extends BaseModal {
           contentType: 'application/json',
         })
           .done((data) => {
-            this.updatePageState('pending');
             this.actionBtn.render();
             this.purchase.set(this.purchase.parse(data));
-            this.payment.render();
+            this.payment = this.createChild(Payment, {
+              balanceRemaining: this.purchase.get('amount'),
+              paymentAddress: this.purchase.get('paymentAddress'),
+              orderId: this.purchase.get('orderId'),
+              isModerated: !!this.order.get('moderator'),
+            });
+            this.listenTo(this.payment, 'walletPaymentComplete',
+              (pmtCompleteData => this.completePurchase(pmtCompleteData)));
+            this.$('.js-pending').append(this.payment.render().el);
+            this.updatePageState('pending');
```

Comment:
The page state needs to be updated before actionBtn is rendered, or it renders the wrong state. 

I think this is what caused #533.